usecases/get_token_metadata.py

```diff
@@ -1,17 +1,28 @@
-from typing import Union
+import json
+from typing import Optional
 
-from gateways.token_gateway import TokenGateway
+from gateways.metadata_gateway import MetadataGateway
 
 
 class GetTokenMetadata:
+    """Get token metadata
 
-    def __init__(self, token_gateway: Union[TokenGateway, None] = None) -> None:
-        self.token_gateway = token_gateway or TokenGateway()
+    :deprecated:
+    """
 
-    def get(self, id: str) -> Union[dict, None]:
-        meta = self.token_gateway.get_token_metadata_from_s3(f"{id}.json")
+    def __init__(self, metadata_gateway: Optional[MetadataGateway] = None) -> None:
+        self.metadata_gateway = metadata_gateway or MetadataGateway()
 
+    def get(self, id: str) -> Optional[dict]:
+        meta_raw = self.metadata_gateway.get_dag_metadata(id)
+        if meta_raw is None:
+            return None
+
+        meta = json.loads(meta_raw)
         if meta:
-            return meta.to_dict()
+            data = meta[id]
+            data['nft'] = data.pop('nft_media')
```

Comment:
The plan was to follow the `Before Merge` section of the PR description.
First, create a PR to fully deprecate the old token metadata endpoint.
Then create the new bucket and move all token metadata to the new bucket making the necessary changes.

The idea was to maintain this old endpoint during the transition so that nothing breaks.

Reply:
I still don't understand how this old endpoint will continue working with the old metadata.json.

The json right now does not have `nft_media`, it has `nft`. Then it wouldn't work, right?

Reply:
Yes, but before merge, we will move the data to another bucket with the `nft_media` and anyone that calls the old endpoint will read from the new bucket and this handler will change to the old format.
This way we won't need to update 2 bucket during the transition to the new endpoint